plugins/tricritical/skills/loop/scripts/review_evidence.py

```
@@ -979,11 +1181,47 @@ def _cycle(
+        role = execution["role"]
+        if execution["scope"] != increment_scopes[role]["scope"]:
+            raise witness.EvidenceError(
+                "Tricritical review dispatch scope differs from the frozen increment"
+            )
+    if not isinstance(cycle["retained_scopes"], list):
+        raise witness.EvidenceError("Tricritical retained scopes must be a list")
+    retained_roles: set[str] = set()
+    for value in cycle["retained_scopes"]:
+        raw_role = value.get("role") if isinstance(value, dict) else None
+        if raw_role not in increment_scopes:
+            raise witness.EvidenceError("Tricritical retained scope has an extra role")
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)